esp_commands/src/esp_commands.c

```diff
@@ -0,0 +1,720 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdbool.h>
+#include <stdio.h>
+#include <unistd.h>
+#include "esp_heap_caps.h"
+#include "esp_commands.h"
+#include "esp_commands_internal.h"
+#include "esp_dynamic_commands.h"
+#include "esp_err.h"
+
+/* Default foreground color */
+#define ANSI_COLOR_DEFAULT 39
+
+/* Pointers to the first and last command in the dedicated section.
+ * See linker.lf for detailed information about the section */
+extern esp_command_t _esp_commands_start;
+extern esp_command_t _esp_commands_end;
+
+typedef struct esp_command_sets {
+    esp_command_set_t static_set;
+    esp_command_set_t dynamic_set;
+} esp_command_sets_t;
+
+/** run-time configuration options */
+static esp_commands_config_t s_config = {
```

Comment:
static data, so is there a small potential for issues here if one thread updates config while another is executing commands?